core/src/fetch_stage.rs

```diff
@@ -216,6 +218,7 @@ impl FetchStage {
                     coalesce,
                     true,
                     None,
+                    true, // only staked connections should be voting
```

Comment:
thanks for making vote explicit, sorry for additional code/test changes, hope won't make bp harder